NESTED-VALIDATION.md

```diff
@@ -0,0 +1,102 @@
+This mapping implements a linked list for iterating through the list of owners. Since we'll only have one owner, `Multicall3` (`0xca11bde05977b3631167028862be2a173976ca11` on [Mainnet](https://etherscan.io/address/0xca11bde05977b3631167028862be2a173976ca11) and [Sepolia](https://sepolia.etherscan.io/address/0xca11bde05977b3631167028862be2a173976ca11)), and the `0x01` address is used as the first and last entry in the linked list, we will see the following overrides:
+- `owners[1] -> 0xca11bde05977b3631167028862be2a173976ca11`
+- `owners[0xca11bde05977b3631167028862be2a173976ca11] -> 1`
+
+And we do indeed see these entries:
+
+- **Key:** 0x316a0aac0d94f5824f0b66f5bbe94a8c360a17699a1d3a233aafcf7146e9f11c <br/>
+  **Value:** 0x0000000000000000000000000000000000000000000000000000000000000001 <br/>
+  **Meaning:** This is `owners[0xca11bde05977b3631167028862be2a173976ca11] -> 1`, so the key can be
+    derived from `cast index address 0xca11bde05977b3631167028862be2a173976ca11 2`.
+
+- **Key:** 0xe90b7bceb6e7df5418fb78d8ee546e97c83a08bbccc01a0644d599ccd2a7c2e0 <br/>
+  **Value:** 0x000000000000000000000000ca11bde05977b3631167028862be2a173976ca11 <br/>
+  **Meaning:** This is `owners[1] -> 0xca11bde05977b3631167028862be2a173976ca11`, so the key can be
+    derived from `cast index address 0x0000000000000000000000000000000000000001 2`.
+
+## State Changes
+
+The following state changes related to the nested Safe execution must be seen, either for the
+Security Council, or the Foundation Safe, depending on which role the simulation was run for:
+
+### `GnosisSafeProxy` - `approvedHashes` mapping update
+
+- **Key:** _Needs to be computed._ <br/>
+  **Before:** `0x0000000000000000000000000000000000000000000000000000000000000000`<br/>
+  **After:** `0x0000000000000000000000000000000000000000000000000000000000000001` <br/>
+
+#### Key Computation
+
+The GnosisSafe `approvedHashes` mapping is updated to indicate approval of this transaction by the chosen Safe Role. The correctness of this slot can be verified as follows:
+- Since this is a nested mapping, we need to use `cast index` twice to confirm that this is the correct slot. The inputs needed are:
+    - The location (`8`) of the `approvedHashes` mapping in the [GnosisSafe storage layout](https://github.com/safe-global/safe-contracts/blob/v1.4.0/contracts/libraries/SafeStorage.sol#L23)
+    - The address of the Safe Role, stored at the env var `$SAFE_ROLE` in the following cast script command.
+    - The safe hash to approve, stored at the env var `$SAFE_HASH` in the following cast script command.
+      It's the value after "Nested hash:" in the simulation output logs.
+- Then using `cast index`, we can compute the key with
+    ```shell
+      $ cast index bytes32 $SAFE_HASH $(cast index address $SAFE_ROLE 8)
+    ```
+    The output of this command must match the key of the state change.
```

Comment:
Is this suggestion accurate? Safe Role was a bit unclear to me
```suggestion
    - The address of the Safe signer, stored at the env var `$SAFE_SIGNER` in the following cast script command.
    - The safe hash to approve, stored at the env var `$SAFE_HASH` in the following cast script command.
      It's the value after "Nested hash:" in the simulation output logs.
- Then using `cast index`, we can compute the key with
    ```shell
      $ cast index bytes32 $SAFE_HASH $(cast index address $SAFE_SIGNER 8)
    ```
    The output of this command must match the key of the state change.
```

Reply:
This is not a blocker, feel free to ignore this change, because I see in https://github.com/ethereum-optimism/superchain-ops/pull/425 we use the same terminology and I don't want to create unnecessary rework

Reply:
I've introduced a definition of `Safe Role` further up in the doc. But I'm happy to change the definition to `Safe Signer`.